### ISSUE TITLE
Expected scrape duration

### ISSUE THREAD

**Issue by FlxPeters** [wrouesnel/postgres_exporter#420]
Hi, i have a question to the expectation about the scrape duration for this exporter. 

We have rolled out several Postgres exporters on our database servers with the following setup:
* The exporter is deployed on the same host next to the Postgres database
* The exporter connects to the database by using a Unix socket
* We use the [queries.yml](https://github.com/wrouesnel/postgres_exporter/blob/master/queries.yaml) from this repo
* `auto-discover-databases` is enabled

We can see a scrape duration from 6 Seconds minimum up to 30 Seconds, which is currently the timeout. 
All above will raise an scrape error. 

So my question is, whats the expected scrape duration for the exporter with the default config. 
Am i doing something wrong or is this pretty long scrape duration normal? 
Do you have any suggestions on how to speed up the metrics scraping? 

Thanks in advance!

**Comment by bennofs** [comment]
I had the same issue, then I found that this is related to the postgresql stats collector not working. Check your postgresql log, if it contains lines like these:

```
prometheus@postgres LOG:  using stale statistics instead of current ones because stats collector is not responding
```

A restart of the postgresql server fixed the issue in my case.

**Comment by FlxPeters** [comment]
Thanks for your feedback. We found and solved the issue with a workaround.
Maybe someone else has the same problem. Here is the solution:

Using `auto-discover-databases` in combination with a socket as host ends up in a log message like this:
`msg="Error opening connection to database (example_db): malformed dsn \"example_db\""`

I'm not sure whats the real problem, but when i use a dedicated user for monitoring connected on `127.0.0.1` the exporter works like charm. We tried multiple approaches using a unix socket like it's described in the docs of [pq](https://pkg.go.dev/github.com/lib/pq?tab=doc), but none of them worked. 

As a feedback for the docs of this project, we may be should add some more examples on configuration.
